### PR TITLE
libc/stdio: Make gets/gets_s work without CONFIG_FILE_STREAM

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

Make gets/gets_s even CONFIG_FILE_STREAM isn't enabled.

## Impact

stdio

## Testing

CI

### CODE REVIEW

File: libs/libc/stdio/lib_libdgets.c [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+          return buf;
+        }
+
+      /* Check for end-of-file */
+
+      else if (ch == EOF)
+        {
+          /* End of file with no data? */
+
+          if (!nch)
```
Comment (pkarashchenko) [inline libs/libc/stdio/lib_libdgets.c:173]:
```suggestion
          if (nch == 0)
```

File: libs/libc/stdio/lib_libdgets.c [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+      /* Check for end-of-line.  This is tricky only in that some
+       * environments may return CR as end-of-line, others LF, and
+       * others both.
+       */
+
+      if (ch == '\n')
+        {
+          /* Convert \r\n to \n */
+
+          if (nch && buf[nch - 1] == '\r')
```
Comment (pkarashchenko) [inline libs/libc/stdio/lib_libdgets.c:149]:
```suggestion
          if (nch > 0 && buf[nch - 1] == '\r')
```

File: libs/libc/stdio/lib_libfgets.c [hunk 3]
```diff
@@ -142,6 +142,13 @@ FAR char *lib_fgets(FAR char *buf, size_t buflen, FILE *stream,
 
       if (ch == '\n')
         {
+          /* Convert \r\n to \n */
+
+          if (nch && buf[nch - 1] == '\r')
```
Comment (pkarashchenko) [inline libs/libc/stdio/lib_libfgets.c:147]:
```suggestion
          if (nch > 0 && buf[nch - 1] == '\r')
```

**Review by pkarashchenko — APPROVED** [review]
Few optional comments

**Comment by xiaoxiang781216** [comment]
Done.

**Review by pkarashchenko — APPROVED** [review]
LGTM!

**Review by anchao — APPROVED** [review]

**Pull request merged**